failed on Configuring with M1 Mac

### System Information

OpenCV version: 4.5.2
[CMakeError.log](https://github.com/opencv/opencv/files/10458589/CMakeError.log)
[CMakeOutput.log](https://github.com/opencv/opencv/files/10458590/CMakeOutput.log)

Operating System / Platform: macOS Monterey 12.5 Apple M1
Compiler & compiler version: XCode 14.0.1
Python version: 3.10
CMake: 3.25.1



### Detailed description

I tried to solve this problem for a week....
```
-- Check size of size_t
CMake Error at /opt/homebrew/Cellar/cmake/3.25.1/share/cmake/Modules/CheckTypeSize.cmake:148 (try_compile):
  Cannot copy output executable

    ''

  to destination specified by COPY_FILE:

    '/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CheckTypeSize/SIZEOF_SIZE_T.bin'

  Recorded try_compile output location doesn't exist:

    /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-2PaoqU/Release/cmTC_7e257.app/cmTC_7e257

Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.25.1/share/cmake/Modules/CheckTypeSize.cmake:278 (__check_type_size_impl)
  3rdparty/libjpeg-turbo/CMakeLists.txt:25 (check_type_size)


-- Check size of unsigned long
CMake Error at /opt/homebrew/Cellar/cmake/3.25.1/share/cmake/Modules/CheckTypeSize.cmake:148 (try_compile):
  Cannot copy output executable

    ''

  to destination specified by COPY_FILE:

    '/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CheckTypeSize/SIZEOF_UNSIGNED_LONG.bin'

  Recorded try_compile output location doesn't exist:

    /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-pJTmFi/Release/cmTC_8bb7b.app/cmTC_8bb7b

Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.25.1/share/cmake/Modules/CheckTypeSize.cmake:278 (__check_type_size_impl)
  3rdparty/libjpeg-turbo/CMakeLists.txt:26 (check_type_size)



-- General configuration for OpenCV 4.5.2 =====================================
--   Version control:               unknown
-- 
--   Platform:
--     Timestamp:                   2023-01-19T16:22:16Z
--     Host:                        Darwin 21.6.0 arm64
--     Target:                      iOS 9.0 arm64
--     CMake:                       3.25.1
--     CMake generator:             Xcode
--     CMake build tool:            /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/xcodebuild_wrapper
--     Xcode:                       14.0.1
--     Configuration:               Debug Release
-- 
--   CPU/HW features:
--     Baseline:                    NEON FP16
--       requested:                 DETECT
-- 
--   C/C++:
--     Built as dynamic libs?:      NO
--     C++ standard:                11
--     C++ Compiler:                /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++  (ver 14.0.0.14000029)
--     C++ flags (Release):         -fembed-bitcode   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG
--     C++ flags (Debug):           -fembed-bitcode   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     C Compiler:                  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
--     C flags (Release):           -fembed-bitcode   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG
--     C flags (Debug):             -fembed-bitcode   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     Linker flags (Release):      
--     Linker flags (Debug):        
--     ccache:                      NO
--     Precompiled headers:         NO
--     Extra dependencies:          -framework UIKit -framework Accelerate -framework CoreGraphics -framework QuartzCore
--     3rdparty dependencies:       libprotobuf libjpeg-turbo libwebp libpng zlib quirc ade
-- 
--   OpenCV modules:
--     To be built:                 calib3d core dnn features2d flann gapi highgui imgcodecs imgproc ml objc objdetect photo stitching video videoio world
--     Disabled:                    -
--     Disabled by dependency:      -
--     Unavailable:                 java python2 python3 ts
--     Applications:                -
--     Documentation:               NO
--     Non-free algorithms:         NO
-- 
--   GUI: 
-- 
--   Media I/O: 
--     ZLib:                        build (ver 1.2.11)
--     JPEG:                        build-libjpeg-turbo (ver 2.0.6-62)
--     WEBP:                        build (ver encoder: 0x020f)
--     PNG:                         build (ver 1.6.37)
--     HDR:                         YES
--     SUNRASTER:                   YES
--     PXM:                         YES
--     PFM:                         YES
-- 
--   Video I/O:
--     AVFoundation:                YES
--     iOS capture:                 YES
-- 
--   Parallel framework:            GCD
-- 
--   Trace:                         YES (built-in)
-- 
--   Other third-party libraries:
--     Custom HAL:                  NO
--     Protobuf:                    build (3.5.1)
-- 
--   Python (for build):            NO
-- 
--   Objective-C wrappers:          YES
-- 
--   Install to:                    /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/install
-- -----------------------------------------------------------------
-- 
-- Configuring incomplete, errors occurred!
See also "/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeOutput.log".
See also "/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeError.log".
============================================================
ERROR: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DFRAMEWORK_NAME=opencv2', '-DIOS_ARCH=arm64', '-DCMAKE_TOOLCHAIN_FILE=/Users/zhaoweiwang/opencv/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/zhaoweiwang/opencv/opencv', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1.
============================================================
Traceback (most recent call last):
  File "/Users/zhaoweiwang/opencv/opencv/platforms/ios/build_framework.py", line 159, in build
    self._build(outdir)
  File "/Users/zhaoweiwang/opencv/opencv/platforms/ios/build_framework.py", line 133, in _build
    self.buildOne(target[0], target[1], main_build_dir, cmake_flags)
  File "/Users/zhaoweiwang/opencv/opencv/platforms/ios/build_framework.py", line 284, in buildOne
    execute(cmakecmd, cwd = builddir)
  File "/Users/zhaoweiwang/opencv/opencv/platforms/apple/cv_build_utils.py", line 13, in execute
    retcode = check_call(cmd, cwd = cwd)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DFRAMEWORK_NAME=opencv2', '-DIOS_ARCH=arm64', '-DCMAKE_TOOLCHAIN_FILE=/Users/zhaoweiwang/opencv/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/zhaoweiwang/opencv/opencv', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1.
```
### Steps to reproduce

```
cd ~/<my_working_directory>
python opencv/platforms/ios/build_framework.py ios
```

run the code above can reproduce

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

Ld /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/Debug/cmTC_189ae.app/cmTC_189ae normal (in target 'cmTC_189ae' from project 'CMAKE_TRY_COMPILE')
    cd /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -target arm64-apple-ios9.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.0.sdk -L/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/build/EagerLinkingTBDs -L/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/Debug -F/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/build/EagerLinkingTBDs -F/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/Debug -filelist /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/build/cmTC_189ae.build/Debug-iphoneos/Objects-normal/arm64/cmTC_189ae.LinkFileList -dead_strip -Xlinker -object_path_lto -Xlinker /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/build/cmTC_189ae.build/Debug-iphoneos/Objects-normal/arm64/cmTC_189ae_lto.o -Xlinker -no_deduplicate -Wl,--as-needed -Wl,-search_paths_first -Xlinker -dependency_info -Xlinker /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/build/cmTC_189ae.build/Debug-iphoneos/Objects-normal/arm64/cmTC_189ae_dependency_info.dat -o /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeTmp/Debug/cmTC_189ae.app/cmTC_189ae
ld: unknown option: --as-needed
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **

CompileC /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/build/cmTC_66127.build/Debug-iphoneos/Objects-normal/arm64/CheckIncludeFile.o /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/CheckIncludeFile.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'cmTC_66127' from project 'CMAKE_TRY_COMPILE')
    cd /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c -target arm64-apple-ios9.0 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-implicit-fallthrough -DCMAKE_INTDIR\=\"Debug-iphoneos\" -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.0.sdk -fstrict-aliasing -Wdeprecated-declarations -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -I/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/Debug/include -I/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/build/cmTC_66127.build/Debug-iphoneos/DerivedSources-normal/arm64 -I/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/build/cmTC_66127.build/Debug-iphoneos/DerivedSources/arm64 -I/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/build/cmTC_66127.build/Debug-iphoneos/DerivedSources -F/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/Debug -fembed-bitcode -fsigned-char -W -Wall -Werror\=return-type -Werror\=non-virtual-dtor -Werror\=address -Werror\=sequence-point -Wformat -Werror\=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body -fvisibility\=hidden -fvisibility-inlines-hidden -fPIE -MMD -MT dependencies -MF /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/build/cmTC_66127.build/Debug-iphoneos/Objects-normal/arm64/CheckIncludeFile.d --serialize-diagnostics /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/build/cmTC_66127.build/Debug-iphoneos/Objects-normal/arm64/CheckIncludeFile.dia -c /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/CheckIncludeFile.c -o /Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/build/cmTC_66127.build/Debug-iphoneos/Objects-normal/arm64/CheckIncludeFile.o
/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-lMCldY/CheckIncludeFile.c:1:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.

/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-UU3Arp/OFF64_T.c:32:12: error: use of undeclared identifier 'off64_t'
  ('0' + ((SIZE / 10)%10)),
           ^
/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-UU3Arp/OFF64_T.c:27:22: note: expanded from macro 'SIZE'
#define SIZE (sizeof(off64_t))
                     ^
/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-UU3Arp/OFF64_T.c:33:12: error: use of undeclared identifier 'off64_t'
  ('0' +  (SIZE    % 10)),
           ^
/Users/zhaoweiwang/opencv/ios/build/build-arm64-iphoneos/CMakeFiles/CMakeScratch/TryCompile-UU3Arp/OFF64_T.c:27:22: note: expanded from macro 'SIZE'
#define SIZE (sizeof(off64_t))
                     ^
5 errors generated.

I hope this can help.
When I running very basic command without additional paramaters from the [guide](https://github.com/opencv/opencv/tree/4.x/platforms/apple)
`python opencv/platforms/apple/build_xcframework.py --out ./build_xcframework`

I got following (full terminal log)

❯ python3 build_xcframework.py --out ./opencv2_xcframework
Using iPhoneOS ARCHS=armv7,arm64
Using iPhoneSimulator ARCHS=x86_64,arm64
Using MacOS ARCHS=x86_64,arm64
Using Catalyst ARCHS=x86_64,arm64
============================================================
Building iPhoneOS frameworks
============================================================
['python3', '/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/build_framework.py', './opencv2_xcframework/iphoneos', '--iphoneos_archs', 'armv7,arm64', '--framework_name', 'opencv2', '--build_only_specified_archs']
Executing: ['python3', '/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/build_framework.py', './opencv2_xcframework/iphoneos', '--iphoneos_archs', 'armv7,arm64', '--framework_name', 'opencv2', '--build_only_specified_archs'] in /Users/andreipalonski/Downloads/opencv-4.7.0/platforms/apple
Executing: python3 /Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/build_framework.py ./opencv2_xcframework/iphoneos --iphoneos_archs armv7,arm64 --framework_name opencv2 --build_only_specified_archs
Using IPHONEOS_DEPLOYMENT_TARGET=9.0
Using iPhoneOS ARCHS=['armv7', 'arm64']
Using iPhoneSimulator ARCHS=None

=================================
CMake
=================================

Executing: ['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DFRAMEWORK_NAME=opencv2', '-DIOS_ARCH=armv7', '-DCMAKE_TOOLCHAIN_FILE=/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/andreipalonski/Downloads/opencv-4.7.0', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode'] in /Users/andreipalonski/Downloads/opencv-4.7.0/platforms/apple/opencv2_xcframework/iphoneos/build/build-armv7-iphoneos
Executing: cmake -GXcode -DAPPLE_FRAMEWORK=ON -DCMAKE_INSTALL_PREFIX=install -DCMAKE_BUILD_TYPE=Release -DOPENCV_INCLUDE_INSTALL_PATH=include -DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty -DFRAMEWORK_NAME=opencv2 -DIOS_ARCH=armv7 -DCMAKE_TOOLCHAIN_FILE=/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake -DCPU_BASELINE=DETECT /Users/andreipalonski/Downloads/opencv-4.7.0 -DCMAKE_C_FLAGS=-fembed-bitcode -DCMAKE_CXX_FLAGS=-fembed-bitcode
-- Setting up iPhoneOS toolchain for IOS_ARCH='armv7'
-- iPhoneOS toolchain loaded
-- Setting up iPhoneOS toolchain for IOS_ARCH='armv7'
-- iPhoneOS toolchain loaded
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- The C compiler identification is AppleClang 14.0.0.14000029
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Detecting C compile features
-- Detecting C compile features - done
-- ocv_init_download: OpenCV source tree is not fetched as git repository. 3rdparty resources will be downloaded from github.com by default.
-- Detected processor: armv7
-- Looking for ccache - not found
-- Performing Test HAVE_CXX_FSIGNED_CHAR
-- Performing Test HAVE_CXX_FSIGNED_CHAR - Failed
-- Performing Test HAVE_C_FSIGNED_CHAR
-- Performing Test HAVE_C_FSIGNED_CHAR - Failed
-- Performing Test HAVE_CXX_W
-- Performing Test HAVE_CXX_W - Failed
-- Performing Test HAVE_C_W
-- Performing Test HAVE_C_W - Failed
-- Performing Test HAVE_CXX_WALL
-- Performing Test HAVE_CXX_WALL - Failed
-- Performing Test HAVE_C_WALL
-- Performing Test HAVE_C_WALL - Failed
-- Performing Test HAVE_CXX_WRETURN_TYPE
-- Performing Test HAVE_CXX_WRETURN_TYPE - Failed
-- Performing Test HAVE_C_WRETURN_TYPE
-- Performing Test HAVE_C_WRETURN_TYPE - Failed
-- Performing Test HAVE_CXX_WNON_VIRTUAL_DTOR
-- Performing Test HAVE_CXX_WNON_VIRTUAL_DTOR - Failed
-- Performing Test HAVE_C_WNON_VIRTUAL_DTOR
-- Performing Test HAVE_C_WNON_VIRTUAL_DTOR - Failed
-- Performing Test HAVE_CXX_WADDRESS
-- Performing Test HAVE_CXX_WADDRESS - Failed
-- Performing Test HAVE_C_WADDRESS
-- Performing Test HAVE_C_WADDRESS - Failed
-- Performing Test HAVE_CXX_WSEQUENCE_POINT
-- Performing Test HAVE_CXX_WSEQUENCE_POINT - Failed
-- Performing Test HAVE_C_WSEQUENCE_POINT
-- Performing Test HAVE_C_WSEQUENCE_POINT - Failed
-- Performing Test HAVE_CXX_WFORMAT
-- Performing Test HAVE_CXX_WFORMAT - Failed
-- Performing Test HAVE_C_WFORMAT
-- Performing Test HAVE_C_WFORMAT - Failed
-- Performing Test HAVE_CXX_WFORMAT_SECURITY
-- Performing Test HAVE_CXX_WFORMAT_SECURITY - Failed
-- Performing Test HAVE_C_WFORMAT_SECURITY
-- Performing Test HAVE_C_WFORMAT_SECURITY - Failed
-- Performing Test HAVE_CXX_WMISSING_DECLARATIONS
-- Performing Test HAVE_CXX_WMISSING_DECLARATIONS - Failed
-- Performing Test HAVE_C_WMISSING_DECLARATIONS
-- Performing Test HAVE_C_WMISSING_DECLARATIONS - Failed
-- Performing Test HAVE_CXX_WMISSING_PROTOTYPES
-- Performing Test HAVE_CXX_WMISSING_PROTOTYPES - Failed
-- Performing Test HAVE_C_WMISSING_PROTOTYPES
-- Performing Test HAVE_C_WMISSING_PROTOTYPES - Failed
-- Performing Test HAVE_CXX_WSTRICT_PROTOTYPES
-- Performing Test HAVE_CXX_WSTRICT_PROTOTYPES - Failed
-- Performing Test HAVE_C_WSTRICT_PROTOTYPES
-- Performing Test HAVE_C_WSTRICT_PROTOTYPES - Failed
-- Performing Test HAVE_CXX_WUNDEF
-- Performing Test HAVE_CXX_WUNDEF - Failed
-- Performing Test HAVE_C_WUNDEF
-- Performing Test HAVE_C_WUNDEF - Failed
-- Performing Test HAVE_CXX_WINIT_SELF
-- Performing Test HAVE_CXX_WINIT_SELF - Failed
-- Performing Test HAVE_C_WINIT_SELF
-- Performing Test HAVE_C_WINIT_SELF - Failed
-- Performing Test HAVE_CXX_WPOINTER_ARITH
-- Performing Test HAVE_CXX_WPOINTER_ARITH - Failed
-- Performing Test HAVE_C_WPOINTER_ARITH
-- Performing Test HAVE_C_WPOINTER_ARITH - Failed
-- Performing Test HAVE_CXX_WSHADOW
-- Performing Test HAVE_CXX_WSHADOW - Failed
-- Performing Test HAVE_C_WSHADOW
-- Performing Test HAVE_C_WSHADOW - Failed
-- Performing Test HAVE_CXX_WSIGN_PROMO
-- Performing Test HAVE_CXX_WSIGN_PROMO - Failed
-- Performing Test HAVE_C_WSIGN_PROMO
-- Performing Test HAVE_C_WSIGN_PROMO - Failed
-- Performing Test HAVE_CXX_WUNINITIALIZED
-- Performing Test HAVE_CXX_WUNINITIALIZED - Failed
-- Performing Test HAVE_C_WUNINITIALIZED
-- Performing Test HAVE_C_WUNINITIALIZED - Failed
-- Performing Test HAVE_CXX_WNO_DELETE_NON_VIRTUAL_DTOR
-- Performing Test HAVE_CXX_WNO_DELETE_NON_VIRTUAL_DTOR - Failed
-- Performing Test HAVE_C_WNO_DELETE_NON_VIRTUAL_DTOR
-- Performing Test HAVE_C_WNO_DELETE_NON_VIRTUAL_DTOR - Failed
-- Performing Test HAVE_CXX_WNO_UNNAMED_TYPE_TEMPLATE_ARGS
-- Performing Test HAVE_CXX_WNO_UNNAMED_TYPE_TEMPLATE_ARGS - Failed
-- Performing Test HAVE_C_WNO_UNNAMED_TYPE_TEMPLATE_ARGS
-- Performing Test HAVE_C_WNO_UNNAMED_TYPE_TEMPLATE_ARGS - Failed
-- Performing Test HAVE_CXX_WNO_COMMENT
-- Performing Test HAVE_CXX_WNO_COMMENT - Failed
-- Performing Test HAVE_C_WNO_COMMENT
-- Performing Test HAVE_C_WNO_COMMENT - Failed
-- Performing Test HAVE_CXX_FDIAGNOSTICS_SHOW_OPTION
-- Performing Test HAVE_CXX_FDIAGNOSTICS_SHOW_OPTION - Failed
-- Performing Test HAVE_C_FDIAGNOSTICS_SHOW_OPTION
-- Performing Test HAVE_C_FDIAGNOSTICS_SHOW_OPTION - Failed
-- Performing Test HAVE_CXX_QUNUSED_ARGUMENTS
-- Performing Test HAVE_CXX_QUNUSED_ARGUMENTS - Failed
-- Performing Test HAVE_C_QUNUSED_ARGUMENTS
-- Performing Test HAVE_C_QUNUSED_ARGUMENTS - Failed
-- Performing Test HAVE_CXX_WNO_SEMICOLON_BEFORE_METHOD_BODY
-- Performing Test HAVE_CXX_WNO_SEMICOLON_BEFORE_METHOD_BODY - Failed
-- Performing Test HAVE_C_WNO_SEMICOLON_BEFORE_METHOD_BODY
-- Performing Test HAVE_C_WNO_SEMICOLON_BEFORE_METHOD_BODY - Failed
-- Performing Test HAVE_CXX_MFPU_VFPV3
-- Performing Test HAVE_CXX_MFPU_VFPV3 - Failed
-- VFPV3 is not supported by C++ compiler
-- Performing Test HAVE_CPU_NEON_SUPPORT (check file: cmake/checks/cpu_neon.cpp)
-- Performing Test HAVE_CPU_NEON_SUPPORT - Failed
-- Performing Test HAVE_CXX_MFPU_NEON (check file: cmake/checks/cpu_neon.cpp)
-- Performing Test HAVE_CXX_MFPU_NEON - Failed
-- NEON is not supported by C++ compiler
-- Performing Test HAVE_CPU_FP16_SUPPORT (check file: cmake/checks/cpu_fp16.cpp)
-- Performing Test HAVE_CPU_FP16_SUPPORT - Failed
-- Performing Test HAVE_CXX_MFPU_NEON_FP16_MFP16_FORMAT_IEEE (check file: cmake/checks/cpu_fp16.cpp)
-- Performing Test HAVE_CXX_MFPU_NEON_FP16_MFP16_FORMAT_IEEE - Failed
-- FP16 is not supported by C++ compiler
-- Performing Test HAVE_CPU_BASELINE_FLAGS
-- Performing Test HAVE_CPU_BASELINE_FLAGS - Failed
CMake Error at cmake/OpenCVCompilerOptimizations.cmake:611 (message):
  Compiler doesn't support baseline optimization flags:
Call Stack (most recent call first):
  cmake/OpenCVCompilerOptions.cmake:327 (ocv_compiler_optimization_options)
  CMakeLists.txt:646 (include)


-- Configuring incomplete, errors occurred!
See also "/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/apple/opencv2_xcframework/iphoneos/build/build-armv7-iphoneos/CMakeFiles/CMakeOutput.log".
See also "/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/apple/opencv2_xcframework/iphoneos/build/build-armv7-iphoneos/CMakeFiles/CMakeError.log".
============================================================
ERROR: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DFRAMEWORK_NAME=opencv2', '-DIOS_ARCH=armv7', '-DCMAKE_TOOLCHAIN_FILE=/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/andreipalonski/Downloads/opencv-4.7.0', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1.
============================================================
Traceback (most recent call last):
  File "/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/build_framework.py", line 181, in build
    self._build(outdir)
  File "/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/build_framework.py", line 139, in _build
    self.buildOne(target[0], target[1], main_build_dir, cmake_flags)
  File "/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/build_framework.py", line 306, in buildOne
    execute(cmakecmd, cwd = builddir)
  File "/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/apple/cv_build_utils.py", line 13, in execute
    retcode = check_call(cmd, cwd = cwd)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DFRAMEWORK_NAME=opencv2', '-DIOS_ARCH=armv7', '-DCMAKE_TOOLCHAIN_FILE=/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/andreipalonski/Downloads/opencv-4.7.0', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1.
============================================================
ERROR: Command '['python3', '/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/build_framework.py', './opencv2_xcframework/iphoneos', '--iphoneos_archs', 'armv7,arm64', '--framework_name', 'opencv2', '--build_only_specified_archs']' returned non-zero exit status 1.
============================================================
Traceback (most recent call last):
  File "/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/apple/build_xcframework.py", line 87, in <module>
    execute(command, cwd=os.getcwd())
  File "/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/apple/cv_build_utils.py", line 13, in execute
    retcode = check_call(cmd, cwd = cwd)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['python3', '/Users/andreipalonski/Downloads/opencv-4.7.0/platforms/ios/build_framework.py', './opencv2_xcframework/iphoneos', '--iphoneos_archs', 'armv7,arm64', '--framework_name', 'opencv2', '--build_only_specified_archs']' returned non-zero exit status 1.

I am facing the exact problem when trying to run `build_framework.py ios`. The script passed CMake after the second try. However, I got another error: 
```
.../third_party/opencv-4.7.0/3rdparty/libjpeg-turbo/src/jdhuff.h:77:19: error: invalid token at start
      of a preprocessor expression
#if SIZEOF_SIZE_T == 8 || defined(_WIN64)
```
I think it is related to the original issue from OP log, something wrong when CMake checks `size_t`.

--- a/platforms/ios/cmake/Toolchains/common-ios-toolchain.cmake
+++ b/platforms/ios/cmake/Toolchains/common-ios-toolchain.cmake
@@ -98,8 +98,17 @@ if(NOT DEFINED IPHONEOS_DEPLOYMENT_TARGET)
 endif()
 
 if(NOT __IN_TRY_COMPILE)
-  set(_xcodebuild_wrapper "${CMAKE_BINARY_DIR}/xcodebuild_wrapper")
-  if(NOT EXISTS "${_xcodebuild_wrapper}")
+  set(_xcodebuild_wrapper "")
+  if(NOT (CMAKE_VERSION VERSION_LESS "3.25.1"))  # >= 3.25.1
+    # no workaround is required (#23156)
+  elseif(NOT (CMAKE_VERSION VERSION_LESS "3.25.0"))  # >= 3.25.0 < 3.25.1
+    if(NOT OPENCV_SKIP_MESSAGE_ISSUE_23156)
+      message(FATAL_ERROR "OpenCV: Please upgrade CMake to 3.25.1+. Current CMake version is ${CMAKE_VERSION}. Details: https://github.com/opencv/opencv/issues/23156")
+    endif()
+  else()  # < 3.25.0, apply workaround from #13912
+    set(_xcodebuild_wrapper "${CMAKE_BINARY_DIR}/xcodebuild_wrapper")
+  endif()
+  if(_xcodebuild_wrapper AND NOT EXISTS "${_xcodebuild_wrapper}")
     set(_xcodebuild_wrapper_tmp "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/xcodebuild_wrapper")
     if(NOT DEFINED CMAKE_MAKE_PROGRAM)  # empty since CMake 3.10
       find_program(XCODEBUILD_PATH "xcodebuild")
@@ -119,7 +128,9 @@ if(NOT __IN_TRY_COMPILE)
     configure_file("${CMAKE_CURRENT_LIST_DIR}/xcodebuild_wrapper.in" "${_xcodebuild_wrapper_tmp}" @ONLY)
     file(COPY "${_xcodebuild_wrapper_tmp}" DESTINATION ${CMAKE_BINARY_DIR} FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
   endif()
-  set(CMAKE_MAKE_PROGRAM "${_xcodebuild_wrapper}" CACHE INTERNAL "" FORCE)
+  if(_xcodebuild_wrapper)
+    set(CMAKE_MAKE_PROGRAM "${_xcodebuild_wrapper}" CACHE INTERNAL "" FORCE)
+  endif()
 endif()
 
 # Standard settings

